Deploy-CAM.ps1

```diff
@@ -2709,6 +2661,155 @@ function Write-Host-Warning() {
     Write-Host ("`n$message") -ForegroundColor Red
 }
 
+# Prompt update Radius Settings in the Keyvault
+function Set-RadiusSettings() {
+    Param(
+        [string]$VaultName
+    )
+    # Check current MFA settings
+    $isRadiusMfaEnabled = Get-AzureKeyVaultSecret `
+        -VaultName $VaultName `
+        -Name "enableRadiusMfa" `
+        -ErrorAction stop
+    $isRadiusMfaEnabled = ([bool]$isRadiusMfaEnabled.SecretValueText)
+
+    $currentRadiusHost = Get-AzureKeyVaultSecret `
+        -VaultName $VaultName `
+        -Name "radiusServerHost" `
+        -ErrorAction stop
+    $currentRadiusHost = ([string]$currentRadiusHost.SecretValueText)
+
+    $currentRadiusPort = Get-AzureKeyVaultSecret `
+        -VaultName $VaultName `
+        -Name "radiusServerPort" `
+        -ErrorAction stop
+    $currentRadiusPort = ([string]$currentRadiusPort.SecretValueText)
+
+    $currentRadiusSecret = Get-AzureKeyVaultSecret `
+        -VaultName $VaultName `
+        -Name "radiusSharedSecret" `
+        -ErrorAction stop
+    $currentRadiusSecret = ([string]$currentRadiusSecret.SecretValueText)
+
+    # Prompt for RADIUS configuration if RADIUS has not been already explicitly been disabled
+    if ($ignorePrompts -and ($enableRadiusMfa -eq $null)) {
```

Comment:
Is enableRadiusMfa a global variable?

Reply:
This setting it to stuff and then checking for $null seems a bit dodgy - can we use more Boolean like logic or is there three states we're worrying about?

Reply:
Yeah it's global and yeah it's three states (enabled, disable, unset)

Reply:
Can we do this without a global variable? Three states is of course fine if there are three states, but looking at the code it seemed to be 'check something for $null that may or may not exist' which is easy to get wrong in PowerShell and cause weird bugs. I've done that a few times.

Reply:
sure, no more globals